ztools/x509/x509.go

```diff
@@ -625,6 +637,44 @@ type Certificate struct {
 	SignedCertificateTimestampList []*ct.SignedCertificateTimestamp
 }
 
+type NoticeNumber []int
+
+type GeneralSubtreeString struct {
+	Data string
+	Max  int
```

Comment:
From RFC5280:
>    "Within this profile, the minimum and maximum fields are not used with
>    any name forms, thus, the minimum MUST be zero, and maximum MUST be
>    absent.  However, if an application encounters a critical name
>    constraints extension that specifies other values for minimum or
>    maximum for a name form that appears in a subsequent certificate, the
>    application MUST either process these fields or reject the
>    certificate."


From X.509:

> The minimum field specifies the upper bound of the area within the subtree. All names whose final name component is above the level specified are not contained within the area. A value of minimum equal to zero (the default) corresponds to the base, i.e., the top node of the subtree. For example, if minimum is set to one, then the subtree excludes the base node but includes subordinate nodes.
> 
> The maximum field specifies the lower bound of the area within the subtree. All names whose last component is below the level specified are not contained within the area. A value of maximum of zero corresponds to the base, i.e., the top of the subtree. An absent maximum component indicates that no lower limit should be imposed on the area within the subtree. For example, if maximum is set to one, then the subtree excludes all nodes except the subtree base and its immediate subordinates. 

Essentially, they have something to do with depth in the subtree, but they "aren't used" in the internet pkix.